Core/Core/Models/DynamicBase.cs

```diff
@@ -203,37 +204,40 @@ public static IEnumerable<PropertyInfo> GetInstanceMembers(Type t)
     /// Gets the names of the typed and dynamic properties that don't have a [SchemaIgnore] attribute.
     /// </summary>
     /// <returns></returns>
-    public IEnumerable<string> GetMemberNames()
-    {
-      var names = new List<string>(properties.Count);
-      foreach (var kvp in properties) names.Add(kvp.Key);
-
-      PopulatePropInfoCache(GetType());
-      var pinfos = propInfoCache[GetType()].Where(x => !x.IsDefined(typeof(SchemaIgnore), true)
-                    && !x.IsDefined(typeof(ObsoleteAttribute), true));
-
-      foreach (var pinfo in pinfos) names.Add(pinfo.Name);
-
-      return names;
-    }
-
+    [Obsolete("Use GetMembers().Keys instead")]
+    public IEnumerable<string> GetMemberNames() => GetMembers().Keys;
+    
     /// <summary>
-    ///  Gets the typed and dynamic properties that don't have a [SchemaIgnore] attribute.
+    ///  Gets the typed and dynamic properties. 
     /// </summary>
-    /// <returns></returns>
-    public Dictionary<string, object> GetMembers()
+    /// <param name="includeMembers">Specifies which members should be included in the resulting dictionary. Can be concatenated with "|"</param>
+    /// <returns>A dictionary containing the key's and values of the object.</returns>
+    public Dictionary<string, object> GetMembers(DynamicBaseMemberType includeMembers = DynamicBaseMemberType.Instance | DynamicBaseMemberType.Dynamic)
```

Comment:
Most critical change is here ☝🏼 and it's non-breaking. All unit test's passed in their original implementation; and also after fixing some non-related issues in them.